File: lib/import-review/platform-saas-hosts.ts
/**
 * Booking / SaaS hosts: may yield booking_url (and tenant-matched services),
 * but must never become the card's phone / email / Instagram / website identity.
 *
 * Keep in sync with scripts/business-enrich/platform_saas_hosts.py.
 */

const PLATFORM_SAAS_HOSTS = [
  "dikidi.net",
  "dikidi.app",
  "support.dikidi.app",
  "glossgenius.com",
  "fresha.com",
  "vagaro.com",
  "booksy.com",
  "mindbodyonline.com",
  "mindbody.io",
  "calendly.com",
  "setmore.com",
  "squareup.com",
  "square.site",
  "book.squareup.com",
  "acuityscheduling.com",
  "styleseat.com",
  "schedulicity.com",
  "gentlemint.com",
  "treatwell.com",
  "salonized.com",
  "phorest.com",
  "timely.com",
  "booker.com",
  "boulevard.io",
];

function hostOnly(raw: string | null | undefined): string {
  const value = (raw || "").trim().toLowerCase();
  if (!value) return "";
  let host = value;
  try {
    if (value.includes("://") || value.startsWith("//") || value.includes("/")) {
      const withProto = value.includes("://")
        ? value
        : `https://${value.replace(/^\/+/, "")}`;
      host = new URL(withProto).hostname.toLowerCase();
    }
  } catch {
    host = value.split("/")[0] || "";
  }
  return host.replace(/^www\./, "");
}

/** True when URL/host is a shared booking SaaS platform. */
export function isPlatformSaasHost(raw: string | null | undefined): boolean {
  const host = hostOnly(raw);
  if (!host) return false;
  return PLATFORM_SAAS_HOSTS.some(
    (h) => host === h || host.endsWith(`.${h}`),
  );
}

/** Prefer storing SaaS URLs as booking_url, never as website. */
export function bookingUrlFromMaybeSaas(
  url: string | null | undefined,
): string | null {
  const trimmed = (url || "").trim();
  if (!trimmed) return null;
  if (!isPlatformSaasHost(trimmed)) return null;
  return trimmed.startsWith("http") ? trimmed : `https://${trimmed}`;
}

File: scripts/business-enrich/platform_saas_hosts.py
#!/usr/bin/env python3
"""Booking / SaaS hosts — mirror of lib/import-review/platform-saas-hosts.ts.

May yield booking_url (and tenant-matched services). Must never become the
card's phone / email / Instagram / website identity.

Keep in sync with lib/import-review/platform-saas-hosts.ts.
"""

from __future__ import annotations

from urllib.parse import urlparse

PLATFORM_SAAS_HOSTS: tuple[str, ...] = (
    "dikidi.net",
    "dikidi.app",
    "support.dikidi.app",
    "glossgenius.com",
    "fresha.com",
    "vagaro.com",
    "booksy.com",
    "mindbodyonline.com",
    "mindbody.io",
    "calendly.com",
    "setmore.com",
    "squareup.com",
    "square.site",
    "book.squareup.com",
    "acuityscheduling.com",
    "styleseat.com",
    "schedulicity.com",
    "gentlemint.com",
    "treatwell.com",
    "salonized.com",
    "phorest.com",
    "timely.com",
    "booker.com",
    "boulevard.io",
)


def host_only(raw: str | None) -> str:
    value = (raw or "").strip().lower()
    if not value:
        return ""
    if "://" in value or value.startswith("//") or "/" in value:
        candidate = value if "://" in value else f"https://{value.lstrip('/')}"
        try:
            value = (urlparse(candidate).hostname or "").lower()
        except ValueError:
            return ""
    return value[4:] if value.startswith("www.") else value


def is_platform_saas_host(raw: str | None) -> bool:
    host = host_only(raw)
    if not host:
        return False
    return any(host == h or host.endswith(f".{h}") for h in PLATFORM_SAAS_HOSTS)


def booking_url_from_maybe_saas(url: str | None) -> str | None:
    trimmed = (url or "").strip()
    if not trimmed or not is_platform_saas_host(trimmed):
        return None
    return trimmed if trimmed.startswith("http") else f"https://{trimmed}"
